mdtraj/geometry/distance.py

```diff
@@ -135,21 +135,28 @@ def compute_displacements(traj, atom_pairs, periodic=True, opt=True):
     return _displacement(xyz, pairs)
 
 
-def compute_center_of_mass(traj):
+def compute_center_of_mass(traj, select='all'):
     """Compute the center of mass for each frame.
 
     Parameters
     ----------
     traj : Trajectory
         Trajectory to compute center of mass for
+    select : str, optional, default=all
+        a mdtraj.Topology selection string that
+        defines the set of atoms of which to calculate
+        the center of mass, the default is all atoms
 
     Returns
     -------
     com : np.ndarray, shape=(n_frames, 3)
          Coordinates of the center of mass for each frame
     """
 
-    com = np.zeros((traj.n_frames, 3))
+    atoms_of_interest = traj.topology.select(select)
+    traj = traj.atom_slice(atoms_of_interest)
```

Comment:
It might be more efficient not to actually slice the trajectory, but to instead just select out the right atoms using list comprehensions and numpy indexing?

Reply:
I see

by using timeit I figured out that this should be the fastest solution specially when nothing is selected (when something is a quite big overhead, as expected)

```
#selection
python -m timeit -s 'import mdtraj; traj=mdtraj.load("tests/data/2EQQ.pdb")' -n 100 'mdtraj.geometry.distance.compute_center_of_mass(traj, "residue 1")'
100 loops, best of 5: 23 msec per loop

#no selection
python -m timeit -s 'import mdtraj; traj=mdtraj.load("tests/data/2EQQ.pdb")' -n 100 'mdtraj.geometry.distance.compute_center_of_mass(traj)'
100 loops, best of 5: 1.04 msec per loop
```

The only problem is that it's much less readable than before

```
def compute_center_of_mass(traj, select=None):
    com = np.empty((traj.n_frames, 3))

    if select is None:
        masses = np.array([a.element.mass for a in traj.top.atoms])
        masses /= masses.sum()

        xyz = traj.xyz

    else:
        atoms_of_interest = traj.topology.select(select)

        masses = np.array([a.element.mass for a in traj.top.atoms if a.index in atoms_of_interest])
        masses /= masses.sum()

        xyz = np.array(traj.xyz[:, atoms_of_interest], order='C')


    for i, x in enumerate(xyz):
        com[i, :] = x.astype('float64').T.dot(masses)
    return com
```

what do you think about it?

Reply:
I think that's pretty readable. Looks good.